### ISSUE TITLE
Regex and UniStr

### ISSUE THREAD

**Issue by bkamins** [JuliaString/Strs.jl#63]
Working with regexes on `UniStr` will be slow as currently all has to be converted to `String` to work (regex as well as the string in which we look for it).

**Comment by ScottPJones** [comment]
Yes, that's only for the moment.  For `ASCIIStr` and `LatinStr`, the same library can be used, and will be faster than using UTF-8, it's a matter of a flag passed to the library.  For regex on `UCS2Str` (and `_UCS2Str`), and `UTF16Str`, those will need version of the PCRE library compiled for 2 byte, and `UTF32Str` will need it compiled for 4 byte.
For the moment, as long as it is working, I'm happy, I'll deal with the performance issues of regexes later, after everything else is working correctly.

**Comment by bkamins** [comment]
Agreed. I just wanted to leave it open as it is an end-user critical part that will require changes to both: how "Regex" object is constructed and how it is called.

**Comment by ScottPJones** [comment]
BTW, I really appreciate your thoughtful reviews and ideas!

**Comment by ScottPJones** [comment]
If you have a chance, please take a look at the Regex support that I just added.
For now, it's just for Str types where the codeunit is `UInt8`.
